src/librustc_trait_selection/traits/project.rs

```diff
@@ -1483,7 +1483,7 @@ fn confirm_impl_candidate<'cx, 'tcx>(
     if substs.len() != tcx.generics_of(assoc_ty.item.def_id).count() {
         tcx.sess
             .delay_span_bug(DUMMY_SP, "impl item and trait item have different parameter counts");
```

Comment:
Could there be another method on tcx that allows specifying a message to avoid the double `delay_span_bug` calls?